storage/tianmu/handler/tianmu_handler.cpp

```diff
@@ -1219,6 +1219,17 @@ ha_rows TianmuHandler::records_in_range([[maybe_unused]] uint inx, [[maybe_unuse
 int TianmuHandler::create(const char *name, TABLE *table_arg, [[maybe_unused]] HA_CREATE_INFO *create_info) {
   DBUG_ENTER(__PRETTY_FUNCTION__);
   try {
+
+    //fix issue 487: bug for create table #mysql50#q.q should return failure and actually return success
+    const size_t	table_name_len = strlen(name);
+    {
```

Comment:
how about if the table_name_len is over max size of file name. and what the purpose of `{}` you added.

Reply:
In the issue comment, I have analyzed the whole process. The tablename_to_filename function will make the table name `#mysql50#q.q` empty, and finally the table name after my code modification point = "./lthdt/", and judge that the last character is '/', indicating that the table name is empty, and an error is returned, which is consistent with innodb. Normal table name = "./lthdt/person" "./lthdt/person".

Errors have been returned at other code points when creating table names exceeding the maximum length.